src/integration-test/java/io/aiven/kafka/connect/http/AvroIntegrationTest.java

```diff
@@ -179,16 +175,12 @@ final void testBasicDelivery() throws ExecutionException, InterruptedException {
         for (final Future<RecordMetadata> sendFuture : sendFutures) {
             sendFuture.get();
         }
-        TestUtils.waitForCondition(() -> bodyRecorderHandler.recorderBodies().size() >= expectedBodies.size(),
-                15000,
-                "All requests received by HTTP server"
-        );
-        log.info("Recorded request bodies: {}", bodyRecorderHandler.recorderBodies());
-        assertIterableEquals(expectedBodies, bodyRecorderHandler.recorderBodies());
 
-        log.info("{} HTTP requests were expected, {} were successfully delivered",
-                expectedBodies.size(),
-                bodyRecorderHandler.recorderBodies().size());
+        await("All expected requests received by HTTP server")
+                .atMost(Duration.ofSeconds(15)).pollInterval(Duration.ofMillis(100))
+                .until(() -> bodyRecorderHandler.recorderBodies().size() >= expectedBodies.size());
+
+        assertThat(bodyRecorderHandler.recorderBodies()).containsExactlyInAnyOrderElementsOf(expectedBodies);
```

Comment:
question: the previous check seemed more strict, it reads as if the order mattered. Is this change intentional?

Reply:
Great catch! You are right! Not sure however whether this is required here. But will fix.